bookserver/crud.py

```diff
@@ -153,14 +156,13 @@ async def create_user(user: AuthUserValidator) -> int:
     print(settings.web2py_private_key)
     crypt = CRYPT(key=settings.web2py_private_key, salt=True)
     new_user.password = str(crypt(user.password)[0])
-    res = None
     try:
         async with async_session.begin() as session:
-            res = session.add(new_user)
+            session.add(new_user)
+        return AuthUserValidator.from_orm(new_user)
     except IntegrityError:
         rslogger.error("Failed to add a duplicate user")
```

Comment:
ok, I've convinced myself this is just wrong.

What should be returned from this crud function when you cannot create a new user.  We certainly should not just return the user that was passed. I think we ought to return None here and then have the api endpoint return the appropriate  message.

Reply:
I agree.

Reply:
I would propose that we augment the AuthUserValidator to make sure that  the username is unique, that the course exists, etc... Then an error like this will be caught and returned as a validation error which seems very appropriate and contains an appropriate level of detail to resolve.

Reply:
Yes, I'll push a change that incorporates this at the crud level.  It turns out that doing it at the validator level is impossible because validators are not async.